cpp/test/sg/rf_depth_test.cu

```diff
@@ -73,7 +73,7 @@ class RfClassifierDepthTest : public ::testing::TestWithParam<int> {
       params.split_algo, params.min_samples_leaf, params.min_samples_split,
       params.min_impurity_decrease, params.bootstrap_features, params.bootstrap,
       params.n_trees, params.max_samples, 0, params.split_criterion, false,
-      params.n_streams, false, 128);
+      params.n_streams, true, 128);
```

Comment:
For this test, I saw intermittent failures when experimental backend is used. Test fails because tree training is terminated before `max_depth` is reached. Increasing number of samples seems to fix this issue. Another approach could be to test less-than-or-equal-to against max_depth instead of equal-to.

Reply:
Fixed in the latest commit